### PR TITLE
Improvements to fireturf code

### PR DESCRIPTION
**Pull request by Lumipharon**
## About The Pull Request
Consolidates the two procs for walking through a flaming turf and standing it in when it ticks.

Currently they have two completely different procs  that do different things and have different arbitrary rules.
For example queen and rav are both currently immune to fire when it ticks, but not when walking over it.

This PR assumed #12091 goes through, but I've removed the rav proc here as well for my own sanity.
## Why It's Good For The Game
Inconsistant, confusing behavior bad.
Consistant, logical code good.
## Changelog
:cl:
balance: Fire turfs now have consistant behavior between walking through them and standing in them
/:cl:

### CODE REVIEW

**Comment by yyzsong** [comment]
So queen and rav are no longer immune to fire damage from ticks?

File: code/modules/projectiles/guns/flamer.dm [hunk 1]
```diff
@@ -585,46 +559,31 @@ GLOBAL_LIST_EMPTY(flamer_particles)
...
-	var/fire_mod = get_fire_resist()
-	if(fire_mod <= 0)
-		return
 	if(status_flags & (INCORPOREAL|GODMODE)) //Ignore incorporeal/invul targets
 		return
-	if(hard_armor.getRating("fire") >= 100)
-		to_chat(src, span_warning("Your suit protects you from most of the flames."))
-		adjustFireLoss(rand(0, burnlevel * 0.25)) //Does small burn damage to a person wearing one of the suits.
+	if(hard_armor.getRating(FIRE) >= 100)
+		to_chat(src, span_warning("Your suit protects you from the flames."))
```
Comment (QualityVan) [inline code/modules/projectiles/guns/flamer.dm:569]: This is a moderate buff to surt. If it's intended, you should mention it in the changelog.

**Review by TiviPlus — APPROVED** [review]

File: code/modules/projectiles/guns/flamer.dm (hunk 1 above)
Comment (Lumipharon) [reply code/modules/projectiles/guns/flamer.dm:569]: Minor buff to surt, aye. I'll get around to reworking fire code entirely at some point

**Comment by FIXTHERADBOXPLEASE** [comment]
Surt not blocking all the fire was the result of jank originally anyway.

**Pull request merged**